### PR TITLE
Feature/s3 c 2798 get bucket notification

### PR DESCRIPTION
**Pull request by dora-korpar**
Adds `getConfigXML` static to NotificationConfiguration class for `getBucketNotification` api

### CODE REVIEW

**Comment by dora-korpar** [comment]
@bert-e approve

**Comment by dora-korpar** [comment]
@bert-e wait

File: lib/models/NotificationConfiguration.js
Comment (miniscruff) [inline lib/models/NotificationConfiguration.js]: The logic here will override errors as you move through, additionally even if something does have an error `_config` will be changed in the process. Not a fan of the methods having (in my opinion hidden) side effects either.

Comment (dora-korpar) [reply lib/models/NotificationConfiguration.js]: If there are any errors, `_config` won't be changed. And yeah, the errors will overwrite each other if there are multiple, but since we're trying to return more specific error messages, it doesn't really make sense to combine them, and it doesn't really matter which one gets returned first

Comment (miniscruff) [reply lib/models/NotificationConfiguration.js]: > If there are any errors, `_config` won't be changed. 

Per iteration in the loop yes, but ( and I am not sure this matters ) if the first was valid and the second item in the list was an error then `_config` would be changed and an error returned.

Comment (dora-korpar) [reply lib/models/NotificationConfiguration.js]: oh, you're totally right, didn't even think of that. i'll reset `this._config` if there's an error

Comment (miniscruff) [reply lib/models/NotificationConfiguration.js]: I believe `_id` is also changed as part of the `_parseId` method but don't take my word for it.

Comment (dora-korpar) [reply lib/models/NotificationConfiguration.js]: I'll take a look. Any changes I make to this class are going to be in this PR tho: https://github.com/scality/Arsenal/pull/1231
The diff is in this PR because currently this branch is based off of that one until it gets merged.

Comment (miniscruff) [reply lib/models/NotificationConfiguration.js]: ok no worries, feel free to point me somewhere else if I do that again.

**Review by naren-rajendran — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]
I did not review the code but approving to unblock PR

**Comment by dora-korpar** [comment]
@bert-e reset

**Pull request merged**
